### DISCUSSION TITLE
Memory usage

### DISCUSSION THREAD

**Discussion by raegen** [lerna/lerna#3387] [Q&A]
So, on bigger projects, running build on all the packages (which is definitely not required and avoidable) can bump into FATAL ERROR: Reached heap limit Allocation failed - JavaScript heap out of memory. Now what this indicates to me, is that memory is not freed up between builds - sometimes this is required, in case of interdependent builds, but considering building those on their own, without the independent packages, works without the aforementioned error, it seems everything is kept in-memory through all of the builds. Why is that so? Is it due to some technical limitations or simply not addressed because it's ultimately avoidable by running the builds sequentially?

**Comment by fahslaj** [comment]
Hi @raegen , what is the exact command you are running? Are you using `--parallel` or `--concurrency`? I would suggest trying out the `--concurrency` flag to limit the number of asynchronous processes that Lerna will run. This might mitigate the issue. (see: https://github.com/lerna/lerna/tree/main/core/global-options#--concurrency)
